### ISSUE TITLE
require('electron') can be overridden by "electron" npm module

### ISSUE THREAD

**Issue by nylki** [electron/electron#3708]
Recently I wanted to test how well electron works on fedora 23. It turns out the prebuilt version doesn't seem to work very well. I am getting this error on the `electron-quick-start` repo:

```
➜  electron-quick-start git:(master) npm start

> electron-quick-start@1.0.0 start /home/tom/Programming/electron-quick-start
> electron main.js

App threw an error when running [TypeError: Cannot read property 'start' of undefined]
A JavaScript error occurred in the main process
Uncaught Exception:
TypeError: Cannot read property 'start' of undefined
    at Object.<anonymous> (/home/tom/Programming/electron-quick-start/main.js:7:23)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Object.<anonymous> (/home/tom/Programming/electron-quick-start/node_modules/electron-prebuilt/dist/resources/default_app/main.js:252:23)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)

```

I haven't tried to build electron manually on fedora 23 yet, but just wanted to let you know that the prebuit fails on my side. (x86-64, fedora 23, npm 3.3.12, node 5.1.1, electron 0.35 (as specified in the package.json))

**Comment by zcbenz** [comment]
I'm unable to reproduce on a fresh installed Fedora 23, can you install a `electron-prebuilt` separately and see if it can start?

**Comment by saiberz** [comment]
I have the same error running in OSX 10.11,
```
electron-quick-start]  on git:master ✔︎  ceec7c3 "Fix the deprecation warning",
› npm start
> electron-quick-start@1.0.0 start /Users/fmorales/ReposRCP/electron-quick-start
> electron main.js

App threw an error when running [TypeError: Cannot read property 'start' of undefined]
``` 

I installed electron-prebuilt globally too and got the same error
```
› electron .
App threw an error when running [TypeError: Cannot read property 'start' of undefined]
```

```
› electron -v
v0.35.4
› node -v
v5.0.0
› npm -v
3.3.6
```

P.S. I opened #3760 before find this one

**Comment by nylki** [comment]
@zcbenz I installed electron globally with `npm install -g electron-prebuilt`
Running then `electron .` on the quick-start repo results in the exact error:

```.js
➜  electron-quick-start git:(master) ✗ electron .
App threw an error when running [TypeError: Cannot read property 'start' of undefined]
A JavaScript error occurred in the main process
Uncaught Exception:
TypeError: Cannot read property 'start' of undefined
    at Object.<anonymous> (/home/tom/Downloads/electron-quick-start/main.js:7:23)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Object.<anonymous> (/home/tom/node_modules/lib/node_modules/electron-prebuilt/dist/resources/default_app/main.js:252:23)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
^C
➜  electron-quick-start git:(master) ✗ electron --version
v0.35.4
➜  electron-quick-start git:(master) ✗ npm --version
3.3.12
```

**Comment by nylki** [comment]
The suggestion by @saiberz in #3760 using:
```
const electron = require('electron');
const app = require('app');
const BrowserWindow = require('browser-window')
```

doesn't fix it here.

**Comment by zcbenz** [comment]
Can you insert following code before `main.js` and paste the results?

```javascript
console.log(require('module').globalPaths);
console.log(require('electron'));
```

**Comment by saiberz** [comment]
```
> electron main.js
[ '~/.nvm/versions/node/v5.2.0/lib/node_modules/electron-prebuilt/dist/Electron.app/Contents/Resources/atom.asar/common/api/lib',
  '~/.nvm/versions/node/v5.2.0/lib/node_modules/electron-prebuilt/dist/Electron.app/Contents/Resources/atom.asar/browser/api/lib',
  '~/.nvm/versions/node/v5.2.0/lib/node_modules/electron-prebuilt/dist/Electron.app/Contents/Resources/atom.asar/browser/api/lib/exports' ]
{ [Function: defineProgram] defineProgram: [Circular], version: '0.4.0', argv: [Function] }
```

**Comment by nylki** [comment]
```
[ '/home/tom/node_modules/lib/node_modules/electron-prebuilt/dist/resources/atom.asar/common/api/lib',
  '/home/tom/node_modules/lib/node_modules/electron-prebuilt/dist/resources/atom.asar/browser/api/lib',
  '/home/tom/node_modules/lib/node_modules/electron-prebuilt/dist/resources/atom.asar/browser/api/lib/exports' ]
{ [Function: defineProgram] defineProgram: [Circular], version: '0.4.0', argv: [Function] }
App threw an error when running [TypeError: Cannot read property 'start' of undefined]
```

**Comment by zcbenz** [comment]
I think I have found the root cause , you must have the `electron` npm module installed globally, which happens to override the `require('electron')` call.

**Comment by saiberz** [comment]
Right , it's working fine now,  I had a node_modules folder with 'electron' module. I thought nvm used to take only libraries installed in my node version, thanks for your help.

**Comment by nylki** [comment]
Works indeed!
Strangely, removing the `electron` module via `npm remove electron -g` did not remove the electron package from my global `node_modules`. I am also using nvm, but thought that only the modules installed in a certain version are active. Atleast, I'd expect removing a module, to not cause any issues. I had to manually remove the `electron` directory in my `node_modules`

**Comment by thebitguru** [comment]
FYI, for other folks that might run into this.  I had accidentally installed `electron` in my home directory (`~/node_modules`).  Even though the home directory was not listed in globalPaths, the version in there was still being used.  Deleting that directory took care of this.

BTW, I am using nvm.

**Comment by dmr** [comment]
```
const electron = require('electron');
const app = require('app');
const BrowserWindow = require('browser-window')
```

Solved the issue for me, thank you :)

**Comment by kevinsimper** [comment]
@dmr Solved it

**Comment by dmr** [comment]
I'm not sure if my idea was a good idea. I thinks I solved this issue by uninstalling the npm package "electron".

"electron" is not the correct package name!
Remove all the electron installations and install "electron-prebuilt", then it works.

**Comment by monmon29** [comment]
I already uninstall my electron both local and global and still it doest not work. I think the problem is in my installation because I already tried it before to the other computer. When I switch to another computer, I got this kind of error.

**Comment by YacheLee** [comment]
I solve this problem by removing all the project folder.
Maybe the folder of node_modules install incomplete or incorrect modules.

**Comment by ViolentCrumble** [comment]
thanks @thebitguru for actually writing what we need to do to fix it...

**Comment by kevinsawicki** [comment]
I'm reopening this and proposing a fix for it in #5662
